assets/examples/upgrade-demo-cluster.sh

```diff
@@ -0,0 +1,171 @@
+# node's opensearch.yml file, and you will need to add entries
+# for the additional nodes in opensearch_dashboards.yml.
+os_node_count=4
+
+# Create Docker network if it doesn't exist. A subnet is defined
+# so that static IP addresses can be assigned to individual continers
+# for TLS certificate SANS purposes.
+create_network()	{
+	docker network create --subnet=172.20.0.0/16 opensearch-dev-net || true
+}	
+
+# Define the TLS certificates for the cluster. This function will
+# generate a root certificate and key, an admin certificate and key,
+# and node certificates for each node in the cluster including
+# OpenSearch Dashboards.
+create_certs()	{
+	# Create the root cert.
+	openssl genrsa -out root-ca-key.pem 2048
+	openssl req -new -x509 -sha256 -key root-ca-key.pem -subj "/C=US/ST=OREGON/L=PORTLAND/O=OPENSEARCH/OU=DOCS/CN=ROOT" -out root-ca.pem -days 730
+
+	# Create the admin cert.
+	openssl genrsa -out admin-key-temp.pem 2048
+	openssl pkcs8 -inform PEM -outform PEM -in admin-key-temp.pem -topk8 -nocrypt -v1 PBE-SHA1-3DES -out admin-key.pem
+	openssl req -new -key admin-key.pem -subj "/C=US/ST=OREGON/L=PORTLAND/O=OPENSEARCH/OU=DOCS/CN=A" -out admin.csr
+	openssl x509 -req -in admin.csr -CA root-ca.pem -CAkey root-ca-key.pem -CAcreateserial -sha256 -out admin.pem -days 730
+
+	# Create OpenSearch node certs.
+	for ((i=1; i <= $os_node_count; i++)); do
+		openssl genrsa -out os-node-0${i}-key-temp.pem 2048
+		openssl pkcs8 -inform PEM -outform PEM -in os-node-0${i}-key-temp.pem -topk8 -nocrypt -v1 PBE-SHA1-3DES -out os-node-0${i}-key.pem
+		openssl req -new -key os-node-0${i}-key.pem -subj "/C=US/ST=OREGON/L=PORTLAND/O=OPENSEARCH/OU=DOCS/CN=os-node-0${i}" -out os-node-0${i}.csr
+		echo "subjectAltName=DNS:os-node-0${i}" | tee -a os-node-0${i}.ext
+		echo "subjectAltName=IP:172.20.0.1${i}" | tee -a os-node-0${i}.ext
+		openssl x509 -req -in os-node-0${i}.csr -CA root-ca.pem -CAkey root-ca-key.pem -CAcreateserial -sha256 -out os-node-0${i}.pem -days 730 -extfile os-node-0${i}.ext
+	done
+
+	# Create OpenSearch Dashboards cert.
+	openssl genrsa -out os-dashboards-01-key-temp.pem 2048
+	openssl pkcs8 -inform PEM -outform PEM -in os-dashboards-01-key-temp.pem -topk8 -nocrypt -v1 PBE-SHA1-3DES -out os-dashboards-01-key.pem
+	openssl req -new -key os-dashboards-01-key.pem -subj "/C=US/ST=OREGON/L=PORTLAND/O=OPENSEARCH/OU=DOCS/CN=os-dashboards-01" -out os-dashboards-01.csr
+	echo 'subjectAltName=DNS:os-dashboards-01' | tee -a os-dashboards-01.ext
+	echo 'subjectAltName=IP:172.20.0.10' | tee -a os-dashboards-01.ext
+	openssl x509 -req -in os-dashboards-01.csr -CA root-ca.pem -CAkey root-ca-key.pem -CAcreateserial -sha256 -out os-dashboards-01.pem -days 730 -extfile os-dashboards-01.ext
+}
+
+# Remove unneeded artifacts from TLS cert creation.
+clean_up_certs()	{
+	rm *temp.pem *csr *ext
+}
+
+
+# Write config files for OpenSearch nodes.
+write_os_configs ()	{
+	for ((i=1; i <= $os_node_count; i++)); do
+		cat <<- EOF > opensearch-0${i}.yml
```

Comment:
`opensearch-0${i}.yml`
Is that extra stuff in the file name intentional? It appears so many other places, it must have significance. But just checking.